libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -1423,6 +1423,8 @@ void AP_GPS::calc_blended_state(void)
             _last_time_updated[i] = state[i].last_gps_time_ms;
         }
     }
+    state[GPS_BLENDED_INSTANCE].last_gps_time_ms = state[best_index].last_gps_time_ms;
```

Comment:
This is already set in line 1149, why would it need to be set here again?

Reply:
yes, on that line it will get written with the correct value.
But calc_blend_weights() is called by line 653,  shortly before, on line 671 function calc_blended_state() gets called. That second function, on line 1280 overwites the written value with zero !!

Hence this problem. Line 1149 is always ignored.

Reply:
> One other possible fix would be to delete line 1303 . But I think that one is a more dangerous one.

That is what seems the correct fix to me, why do you think it is dangerous? If it isn't set in line 1149 then it isn't using blended so it doesn't matter.
In any case, what this line sets is different than what line 1149 sets and that line does it best I think. Also, whatever the fix ends up being, only one setting should stay.

Reply:
It would be nice to get @rmackay9 input on this one liner

Reply:
Technically when a driver is unloaded right now we always memset the state vector, so having a 0 would make sense when the blended instance is unavailable. However the set on 1303 will never be hit if we aren't doing blending, meaning the clear to 0 isn't helping us. I think leaving it set is probably safe however though, so I'm in favour of the delete 1303 approach.